src/core/border_router/infra_if.cpp

```diff
@@ -111,6 +111,34 @@ void InfraIf::HandledReceived(uint32_t aIfIndex, const Ip6::Address &aSource, co
     }
 }
 
+Error InfraIf::DiscoverNat64Prefixes(void)
+{
+    OT_ASSERT(mInitialized);
+
+    return otPlatInfraIfDiscoverNat64Prefixes(mIfIndex);
+}
+
+void InfraIf::DiscoverNat64PrefixesDone(uint32_t aIfIndex, const Ip6::Address *aAddresses, uint8_t aNumAddresses)
+{
+    Error error = kErrorNone;
+
+    OT_UNUSED_VARIABLE(aNumAddresses);
+
+    VerifyOrExit(mInitialized && mIsRunning, error = kErrorInvalidState);
+    VerifyOrExit(aIfIndex == mIfIndex, error = kErrorInvalidArgs);
+    VerifyOrExit(aAddresses != nullptr, error = kErrorInvalidArgs);
```

Comment:
If no NAT64 prefix/addresses is found, we expect the platform to invoke callback with empty array.
I think we should allow `nullptr` for `aAddresses` in such a case (i.e. only when `aNumAddresses == 0`).

Reply:
Updated.